releases/release-5.0.3.md

```diff
@@ -0,0 +1,115 @@
+---
+title: TiDB 5.0.3 Release Notes
+---
+
+# TiDB 5.0.3 Release Notes
+
+Release date: June 30, 2021
+
+TiDB version: 5.0.3
+
+## Compatibility Changes
+
+## New Features
+
+## Improvements
+
++ TiDB
+
+    - Support pushing down the `TopN` operator to TiFlash [#25162](https://github.com/pingcap/tidb/pull/25162)
+    - Support pushing down the built-in function `json_unquote()` to TiKV [#25720](https://github.com/pingcap/tidb/pull/25720)
+    - Support removing the union branch from the dual table [#25614](https://github.com/pingcap/tidb/pull/25614)
+    - Support pushing down the built-in function `replace()` to TiFlash [#25565](https://github.com/pingcap/tidb/pull/25565)
+    - Support pushing down the built-in functions `unix_timestamp()`, `concat()`, `year()`, `day()`, `datediff()`, `datesub()`, `castTimeAsString()`, and `concat_ws()` to TiFlash [#25564](https://github.com/pingcap/tidb/pull/25564)
+    - Change the aggregate operator's cost factor [#25241](https://github.com/pingcap/tidb/pull/25241)
+    - Support pushing down the `Limit` operator to TiFlash [#25159](https://github.com/pingcap/tidb/pull/25159)
+    - Support pushing down the built-in function `str_to_date` to TiFlash [#25148](https://github.com/pingcap/tidb/pull/25148)
+    - Allow the MPP outer join to choose the build table based on the table row count [#25142](https://github.com/pingcap/tidb/pull/25142)
+    - Support pushing down the built-in functions `left()`, `right()`, and `abs()` to TiFlash [#25133](https://github.com/pingcap/tidb/pull/25133)
+    - Support pushing down the Broadcast Cartesian join to TiFlash [#25106](https://github.com/pingcap/tidb/pull/25106)
+    - Support pushing down the `Union All` operator to TiFlash [#25051](https://github.com/pingcap/tidb/pull/25051)
+    - Support balancing the MPP query workload among different TiFlash nodes based on Regions [#24724](https://github.com/pingcap/tidb/pull/24724)
+    - Support invalidating stale Regions in the cache after the MPP query is executed [#24432](https://github.com/pingcap/tidb/pull/24432)
+
++ TiKV
+
+    - Limit the TiCDC sink's memory consumption [#10305](https://github.com/tikv/tikv/pull/10305)
+    - Add the memory-bounded upper limit for the TiCDC old value cache [#10313](https://github.com/tikv/tikv/pull/10313)
+
++ TiFlash
+
+    - Support casting the `STRING` type to the `DOUBLE` type
+    - Support the `STR_TO_DATE` function
+    - Optimize the non-joined data in right outer join using multiple threads
+    - Support the Cartesian join
+    - Support the `LEFT()` and `RIGHT()` functions
+    - Support automatically invalidating stale Regions in MPP queries
+    - Support the `ABS()` function
+
```

Comment:
+ Tools

    - TiCDC

        * Refine GRPC reconnect logic, increase the throughput of kv client [#1922](https://github.com/pingcap/ticdc/pull/1922)
        * Make sorter IO errors more user-friendly [#1977](https://github.com/pingcap/ticdc/pull/1977)